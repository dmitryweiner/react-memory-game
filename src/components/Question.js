import React from "react";
import Countdown360 from "react-countdown360";

const Question = ({numbers, delay, handleQuestionComplete}) => {
  return <>
      <Countdown360
        seconds={delay}
        onComplete={handleQuestionComplete}
      />
      <div data-testid="question-numbers">
        {numbers.join(" ")}
      </div>
    </>;
};

export default Question;
