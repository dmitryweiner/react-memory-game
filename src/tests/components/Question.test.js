import React from "react";
import renderer from "react-test-renderer";
import Question from "../../components/Question";
import { render, getNodeText } from "@testing-library/react";

test("Question calls proper handler when time is over", async () => {
  const props = {
    numbers: [1, 2, 3],
    delay: 1,
    handleQuestionComplete: jest.fn()
  };
  const callback = props.handleQuestionComplete;

  const { getByTestId } = render(<Question {...props} />);
  const numbersElement = getByTestId("question-numbers");
  expect(numbersElement).toBeInTheDocument();
  expect(getNodeText(numbersElement)).toBe(props.numbers.join(" "));

  expect(callback).not.toBeCalled();
  await new Promise((resolve) => setTimeout(resolve, 2000));
  expect(callback).toBeCalled();
  expect(callback).toHaveBeenCalledTimes(1);
});

test("Question", () => {
  const props = {
    numbers: [1, 2, 3],
    delay: 10,
    handleQuestionComplete: jest.fn()
  };
  const component = renderer.create(<Question {...props} />).toJSON();
  expect(component).toMatchSnapshot();
});
