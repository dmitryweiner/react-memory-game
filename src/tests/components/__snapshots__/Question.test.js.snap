// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question 1`] = `
Array [
  <div
    className="sc-fzqNJr gUxgXI"
  >
    <div
      className="sc-fzoyAV iWAzvD"
    >
      10
      <br />
      <div
        className="sc-fzoLag iTKmqL"
      >
        seconds
      </div>
    </div>
  </div>,
  <div
    data-testid="question-numbers"
  >
    1 2 3
  </div>,
]
`;
